### ISSUE TITLE
Where to do the coin change calculation?

### ISSUE THREAD

**Issue by georgescharlesbrain** [juzybits/polymedia-gotbeef#2]
Hi,

I noticed you receive any amount and return the change for your deposits. ([move code block](https://github.com/juzybits/polymedia-gotbeef/blob/4b80c91bd49628d1227afa5140cc8ac2140fdde2/sui/sources/bet.move#L197C1-L204C11))
Have you seen the [talk about PTBs by Alex from Mysten](https://www.youtube.com/live/W8dVZQy-dlI?si=jN_gTqkoxIikkjmS&t=637)?
He claims that best practice is to handle the exact amount calculation and coin splitting in the front end in the previous transaction of the PTB. 
Alex: "It simplifies the code, more visibility and people don't have to worry the move contract will steal the coin." 

What's your opinion? You've been part of that discussion back in the day?

,KR

PS: Thx again for making your code public :).

**Comment by juzybits** [comment]
Thanks for the feedback!

**Comment by juzybits** [comment]
Yes, it is best practice to use Programmable Transaction Blocks, and in fact the Got Beef front-end uses PTBs to send the exact Coin balance to the contract: https://github.com/juzybits/polymedia-gotbeef/blob/main/web/src/js/Fund.tsx#L75-L99

The reason the contract can accept larger balances and handle change is because it was written several months before PTBs were added to Sui (August 2022 vs spring 2023). When PTBs came out, I updated the front-end to use them, but left the contract untouched as it had already been well tested and there was no real drawback in allowing it to handle change (even when in practice the front-end always sends the exact amount).

Btw Got Beef was my 1st ever Sui project (and my 2nd React project) so there may be some things that could have been done better. Always happy to get feedback or pull requests.

Thanks for the feedback!
